### ISSUE TITLE
Delimit url param names by . in addition to /

### ISSUE THREAD

**Issue by didip** [go-chi/chi#167]
Hi chi devs,

Given this pattern `/root/:leaf.:mimeType`...

I expect the following to match:

```
/root/foo.txt:
    :leaf => foo
    :mimeType => txt

/root/bar.json:
    :leaf => bar
    :mimeType => json
```

But Chi fails to parse both `leaf` and `mimeType`. Is that expected?

**Comment by Vektah** [comment]
I don't think even:
`/root/:leaf.json` will do what you expect. 

I think Chi matches prefixes only which means only one wildcard per path segment.

**Comment by didip** [comment]
Hm, bummer.

I understand not wanting to have complex parser and slowing the router down.

But it would be nice if the URL chunk is tokenized by `/`, `.`, and `?` at minimum to avoid surprises for people coming from another frameworks or routers.

**Comment by pkieltyka** [comment]
@didip this is completely possible, and you're right the only thing that would be necessary is to parse the param key up until one of the param delimiters such as `/` or `.`

`?` isn't really an acceptable delimiter in any case for a url route.

One other thought is perhaps we should change the delimiter in generate for params to routes like `/articles/{id}` instead of `/articles/:id` this would free up `:` as a usable delimiter.. cc: @VojtechVitek

**Comment by VojtechVitek** [comment]
`{id}` works for me

**Comment by didip** [comment]
Thanks all for the response!

I am down with either {id} or :id

**Comment by pkieltyka** [comment]
@didip btw, I agree we should improve param-key parsing, but I after thinking a bit more about your use-case which is to have the `.:mimeType`, I don't think it's ideal to add `.:mimeType` to every single one of your routes. It's probably better to write a middleware that will parse the extension from the RoutePath and directly put it into the URLParams on the routing context.

**Comment by c2h5oh** [comment]
I quite like typed parameters Flask has:

- `<string:param_name>` or simply `<param_name>` - match everything but slashes
- `<path:param_name>` - match everything, including slashes
- `<int:param_name>`, `<float:param_name>` - quite obvious
- `<any:enum1,enum2,enum3>`

Flask supports variable routes with different converters (the part before ":"), but that's a bit much for us imo. At the same time supporting at least `path` would be nice.

**Comment by pkieltyka** [comment]
solved by v3 release 02e6bbdfd14640d8c1beb15ec3a38140400fb5c2

See complete CHANGELOG for v3: https://github.com/go-chi/chi/blob/master/CHANGELOG.md
